### PR TITLE
Fix validation uint int sizes

### PR DESCRIPTION
**Pull request by fullkomnun**
## Description

Previously an ABI that contained a numeric type (int/uint) of size that is not a power of 2 (such as uint40, int24) would result in the validator / check function being set on Zod as undefined and lead to a cryptic TypeError during validation.
I have setup validators for all valid numeric type sizes ([u]int8, [u]int 16, ... , [u]int256) and throw an error right away if during conversion of JSON scheme to Zod an unsupported type is encountered. 
Generated 'single parameter' ABI test cases for all valid sizes.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — APPROVED** [review]
LGTM. Thanks for your contribution

File: packages/web3-validator/coverage/coverage-final.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (luu-alex) [inline packages/web3-validator/coverage/coverage-final.json]: think this was generated when formatted, can be removed?

**Review by jdevcs — APPROVED** [review]
Thanks for contribution

File: packages/web3-validator/coverage/coverage-final.json (hunk 1 above)
Comment (fullkomnun) [reply packages/web3-validator/coverage/coverage-final.json]: Yes, I will remove.
`packages/web3-validator/test/config/jest.config.js` was also changed only due to 'prettier' but apparently needed formatting.

**Comment by jdevcs** [comment]
Same in this PR , update is required to fix Conflicting file, I'll merge PR after it, Thanks

**Review by luu-alex — APPROVED** [review]

**Review by jdevcs — COMMENTED** [review]
unit test not passing

**Comment by fullkomnun** [comment]
> unit test not passing

Checking

**Comment by CodeLikeLisa** [comment]
is the solution to use an older version of web3js?

File: packages/web3-validator/src/validator.ts [hunk 2]
```diff
@@ -67,6 +67,10 @@ const convertToZod = (schema: JsonSchema): ZodType => {
 	}
 
 	if (schema?.format) {
+		if (!formats[schema.format]) {
+			throw new Error(`format ${schema.format} is unsupported`);
```
Comment (jdevcs) [inline packages/web3-validator/src/validator.ts]: I re invoked tests and now unit tests are passing, but codecov patch is not, could you add unit test hitting L71, Thanks @fullkomnun

**Comment by luu-alex** [comment]
Hey there im looking into an issue. @fullkomnun theres an inconsistent integration test [failing in the pipelines ](https://github.com/web3/web3.js/actions/runs/6332374631/job/17241991863?pr=6434#step:5:247) and i think it stems from the last [PR ](https://github.com/web3/web3.js/pull/6435) you pushed. The [abi ](https://github.com/web3/web3.js/blob/4.x/packages/web3-eth-contract/test/fixtures/erc721.ts#L129) and the [testcase that is failing](https://github.com/fullkomnun/web3.js/blob/fix_validation_uint_int_sizes/packages/web3-eth-contract/test/integration/local_account/contract_erc721.test.ts#L193). any ideas?

**Review by jdevcs — COMMENTED** [review]
codecov/patch Failing after 1s — 66.66% of diff hit (target 89.88%)

**Comment by luu-alex** [comment]
@luu-alex  update, it seems to be an eventslog issue. not related to PR

File: packages/web3-validator/src/validator.ts (hunk 2 above)
Comment (Muhammad-Altabba) [inline packages/web3-validator/src/validator.ts]: Could you please use a new error code and an error class that inherits from `BaseWeb3Error` similar to other errors inside `web3-errors` package? And mention this in the CHSANGELOG.md as well?

**Comment by jdevcs** [comment]
> codecov/patch Failing after 1s — 66.66% of diff hit (target 89.88%)

@fullkomnun

File: packages/web3-validator/CHANGELOG.md [hunk 3]
```diff
@@ -151,4 +151,6 @@ Documentation:
 
 ## Fixed
 
--   Multi-dimensional arrays are now handled properly when parsing ABIs
+-   Multi-dimensional arrays are now handled properly when parsing ABIs (#6434)
+-   Validator will now properly handle all valid numeric type sizes: intN / uintN where 8 <= N <= 256 and N % 8 == 0 (#6434)
+-   Will now throw SchemaFormatError when unsupported format is given in `convertToZod` method (#6434)
```
Comment (Muhammad-Altabba) [inline packages/web3-validator/CHANGELOG.md]:
```suggestion
-   Will now throw SchemaFormatError when unsupported format is passed to `convertToZod` method (#6434)
```

File: packages/web3-errors/src/errors/schema_errors.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+import { ERR_SCHEMA_FORMAT } from '../error_codes.js';
+import { BaseWeb3Error } from '../web3_error_base.js';
+
+export class SchemaFormatError extends BaseWeb3Error {
+	public code = ERR_SCHEMA_FORMAT;
+
+	public constructor(public format: string) {
+		super(`format ${format} is unsupported`);
```
Comment (Muhammad-Altabba) [inline packages/web3-errors/src/errors/schema_errors.ts]:
```suggestion
	public constructor(public type: string) {
		super(`Format for the type '${type}' is unsupported`);
```

File: packages/web3-errors/test/unit/__snapshots__/errors.test.ts.snap [hunk 5]
```diff
@@ -333,6 +333,16 @@ Object {
 }
 `;
 
+exports[`errors SchemaFormatError should have valid json structure 1`] = `
+Object {
+  "code": 1200,
+  "format": "unsupported",
+  "innerError": undefined,
+  "message": "format unsupported is unsupported",
```
Comment (Muhammad-Altabba) [inline packages/web3-errors/test/unit/__snapshots__/errors.test.ts.snap]:
```suggestion
  "message": "Format for the type 'unsupported' is unsupported",
```

**Review by Muhammad-Altabba — APPROVED** [review]
Looks good, thanks,
I just have suggestions for the error string...

**Review by jdevcs — APPROVED** [review]

**Comment by ngmachado** [comment]
Any news about this PR?

**Comment by luu-alex** [comment]
@ngmachado  will be merged in today

**Pull request merged**